x-pack/test/functional/page_objects/index_management_page.ts

```diff
@@ -41,26 +41,47 @@ export function IndexManagementPageProvider({ getService }: FtrProviderContext)
       });
     },
 
+    async performIndexActionInDetailPanel(action: string) {
+      await this.clickContextMenuInDetailPanel();
+      switch (action) {
+        case 'flush':
+          await testSubjects.click('flushIndexMenuButton');
+      }
+    },
```

Comment:
Yeah I could have done that. But in the future we will be adding more actions and there's 8 of them. I think it's cleaner to add another case rather than a bunch of if elses.

Reply:
Word.  Well, dont forget the `break;` statement when ya add more!  :hahahahaa:

Reply:
> Yeah I could have done that. But in the future we will be adding more actions and there's 8 of them.

You might add a comment to that effect. I had the same thought as @wayneseymour when I first looked at the code.

Reply:
Fixed.